### PR TITLE
paths_to_object supports ?shape=1

### PR DESCRIPTION
**Pull request by will-moore**
This adds `?shape=1` support to paths_to_objects. (Similar to #159 for ROIs).
Required to support ?shape=1 in iviewer.

Test added in https://github.com/ome/openmicroscopy/pull/6239

To test:
 - With a shape ID: go to `/webclient/api/paths_to_object/?shape=2`
 - Should see a 'path' of objects, including image, roi and shape.
 - Using an invalid Shape ID should return empty JSON list.

### CODE REVIEW

File: omeroweb/webclient/show.py [hunk 1]
```diff
@@ -449,10 +449,26 @@ def get_image_ids(conn, datasetId=None, groupId=-1, ownerId=None):
...
+    params = omero.sys.ParametersI()
+    params.addId(shape_id)
+    query = """select roi from Roi roi
+        left outer join roi.shapes as shape
+        where shape.id=:id"""
+    query_service = conn.getQueryService()
+    result = query_service.findByQuery(query, params, conn.SERVICE_OPTS)
+    if result:
+        roi = result
+        return (roi.id.val, roi.image.id.val)
```
Comment (jburel) [inline omeroweb/webclient/show.py:463]: with the method being ``get_image_id_for_shape``, I will expect to have the image Id returned first.

**Comment by jburel** [comment]
Shape on an image in dataset
```
{"paths": [[{"type": "experimenter", "id": 454}, {"type": "dataset", "id": 18469, "childCount": 5}, {"type": "image", "id": 122915}, {"type": "roi", "id": 291069}, {"type": "shape", "id": 296898}]]}
```

Shape on a wellsample
```
{"paths": [[{"type": "experimenter", "id": 454}, {"type": "plate", "id": 13855}, {"type": "well", "id": 20111}, {"type": "roi", "id": 92700}, {"type": "shape", "id": 98379}]]}
```
I think the path is probably misleading in that case since the user need to select the wellsample to view the image

**Comment by will-moore** [comment]
With that last commit, I now see a path to SPW image as:
```
{ "paths": [ [ { "type": "experimenter", "id": 2 }, { "type": "screen", "id": 52 },
{ "type": "plate", "id": 54 },  { "type": "well", "id": 473 }, { "type": "wellsample", "id": 541 },
{ "type": "image", "id": 1793 }, { "type": "roi", "id": 184304 },
{ "type": "shape", "id": 184304 } ] ] }```

**Comment by will-moore** [comment]
Failing test at https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-test-integration/lastCompletedBuild/testReport/OmeroWeb.test.integration.test_show/TestShow/test_well_image/ should be fixed by https://github.com/ome/openmicroscopy/pull/6239/commits/b682fd9b1d104da6dbf93c10000fec8bbb6f2db3

**Comment by jburel** [comment]
* User 1. shape on an image in HCS https://merge-ci.openmicroscopy.org/web/iviewer/?shape=297241
https://merge-ci.openmicroscopy.org/web/webclient/api/paths_to_object/?shape=297241
```
{"paths": [[{"type": "experimenter", "id": 452}, {"type": "dataset", "id": 2719, "childCount": 1}, {"type": "image", "id": 16015}, {"type": "roi", "id": 291362}, {"type": "shape", "id": 297241}]]}
```
This is not what I was expecting

**Comment by will-moore** [comment]
That image is in a Dataset: 
https://merge-ci.openmicroscopy.org/web/webclient/?show=image-16015
The paths_to_object code assumes that if an Image is in a Dataset then don't need to check if it's in a SPW.
To change that assumption would be outside the scope of this PR.

**Comment by jburel** [comment]
ok that makes sense

**Comment by jburel** [comment]
Looking at a "true" well
```
{"paths": [[{"type": "experimenter", "id": 452}, {"type": "screen", "id": 1654}, {"type": "plate", "id": 2506}, {"type": "well", "id": 7755}, {"type": "wellsample", "id": 15505}, {"type": "image", "id": 16016}, {"type": "roi", "id": 291363}, {"type": "shape", "id": 297242}], [{"type": "experimenter", "id": 452}, {"type": "screen", "id": 2505}, {"type": "plate", "id": 2506}, {"type": "well", "id": 7755}, {"type": "wellsample", "id": 15505}, {"type": "image", "id": 16016}, {"type": "roi", "id": 291363}, {"type": "shape", "id": 297242}]]}

```

**Comment by jburel** [comment]
Only the edge case of an image in dataset and a plate is not handled by this PR.

**Comment by will-moore** [comment]
So is this good to merge?

**Comment by jburel** [comment]
The edge case could be fixed in a follow-up PR if required
Good to merge

**Pull request merged**